conf/cache.c

```diff
@@ -102,6 +103,51 @@ M0_INTERNAL bool m0_conf_cache_contains(struct m0_conf_cache *cache,
 	return ret;
 }
 
+M0_INTERNAL struct m0_conf_obj *
+m0_conf_cache_lookup_dynamic_fid(const struct m0_conf_cache *cache,
+				 const struct m0_fid *id)
+{
+	struct m0_conf_obj *obj;
+	uint32_t cnt1;
+	uint32_t cnt2;
+        /*
+         * For process and service fid different approach is taken to compare
+	 * FID's. Specifically for process/service fid considering base fid only
+	 * (excluding counter bit) for comparison. Counter bits are higher 32
+	 * bist of the key.
+	 *  e.g. 
+	 *  |-------- Container------------| |----------------Key-------------|
+	 *
+	 *  F F F F F F F F F F F F F F F F : F F F F F F F F  F F F F F F F F
+	 *                                    |-Counter bits-|
+	 */
+	/*
+	 * TODO: Current implementation we are assuming that dynamic FID's will
+	 * always be generated in sequential manner (w.r.t counter bits), so
+	 * that we can fetch latest last dynamic FID, but It may not be in
+	 * sequential always. This scenario needs to be handle.
+	 */
+	if ((m0_fid_tget(id) == 'r') ||
```

Comment:
Adde check for,
if ( (id->f_key >> 32 & 0xFFFFFFFF) != 0 &&


Also add a macro for geeting dynamic value "d->f_key >> 32 & 0xFFFFFFFF"